File: test/org/lice/parse2/LexCommentTest.java
package org.lice.parse2;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class LexCommentTest {
	@Test(timeout = 100)
	public void testLexComment() {
		String src = "; Simply comments";
		Lexer l = new Lexer(src);
		assertEquals(l.currentToken().getType(), Token.TokenType.EOI);
	}

	@Test(timeout = 100)
	public void testLexComment2() {
		String src = "; Simply comments\n@dentifier";
		Lexer l = new Lexer(src);
		assertEquals(l.currentToken().getType(), Token.TokenType.Identifier);
		assertEquals(l.peekOneToken().getType(), Token.TokenType.EOI);
	}
}

File: test/org/lice/parse/LexCommentTest.kt
package org.lice.parse

import org.junit.Test

import org.junit.Assert.assertEquals

class LexCommentTest {
	@Test
	fun testLexComment() {
		val src = "; Simply comments"
		val l = Lexer(src)
		assertEquals(Token.TokenType.EOI, l.currentToken().type)
	}

	@Test
	fun testLexComment2() {
		val src = "; Simply comments\n@dentifier"
		val l = Lexer(src)
		assertEquals(Token.TokenType.Identifier, l.currentToken().type)
		assertEquals("@dentifier", l.currentToken().strValue)
		assertEquals(Token.TokenType.EOI, l.peekOneToken().type)
	}
}
